libbpf-tools/hardirqs.bpf.c

```diff
@@ -38,12 +39,21 @@ struct {
 
 static struct info zero;
 
+static __always_inline bool is_target_cpu() {
+  if (targ_cpu < 0)
+    return true;
+
+  return (u32)targ_cpu == bpf_get_smp_processor_id();
```

Comment:
targ_cpu is defined as a bool (const volatile bool targ_cpu = -1), how does the whole thing works? Did you do any testing?

Reply:
fixed